code/modules/projectiles/projectile.dm

```diff
@@ -241,12 +243,22 @@
 /**
  * Called when the projectile hits something
  *
- * @params
- * target - thing hit
- * blocked - percentage of hit blocked
- * pierce_hit - are we piercing through or regular hitting
+ * By default parent call will always return [BULLET_ACT_HIT] (unless qdeleted)
+ * so it is save to assume a successful hit in children (though not necessarily successfully damaged - it could've been blocked)
+ *
+ * Arguments
+ * * target - thing hit
+ * * blocked - percentage of hit blocked (0 to 100)
```

Comment:
looking through this code, I've realized i don't really like `blocked` because i think that's why it was confused to be a boolean in the first place, could we try renaming the arg to `blocked_percentage` or something in the spirit of this "comphrensive refactor". ounce of prevention is worth a pound of cure (having to fix people adding booleans as args to this shit again)

Reply:
Out of scope IMO, at least I wouldn't want to do this without also updating `blocked` in `apply_damage`, `apply_effect`, etc. args

Reply:
alright that's fine then